test/integration/test_request_response_api.py

```diff
@@ -43,195 +50,168 @@ def tearDownModule() -> None:
+        data = client.call('transaction.get')['data']
+        ids = [self.trim_msg_id(transaction['_id'])
+               for transaction in data]
 
         with self.subTest():
-            self.assertIn('simple', response['data'])
+            self.assertTrue(
+                set(ids).issubset([str(i) for i in self.transaction_ids]))
         with self.subTest():
-            self.assertIn('example_item', response['data'])
+            self.assertEqual(len(data), 50)
 
-    def test_response_unaltered_by_message_content(self) -> None:
-        response = client.call('db', 'this is ignored')
+    def test_response_can_be_given_new_limit(self) -> None:
+        data = client.call('transaction.get', {'count': 10})['data']
 
         with self.subTest():
-            self.assertIn('simple', response['data'])
-        with self.subTest():
-            self.assertIn('example_item', response['data'])
+            self.assertEqual(len(data), 10)
 
+    def test_limit_cant_be_greater_than_50(self) -> None:
+        response = client.call('transaction.get', {'count': 51})
 
-class TestRouteExampleItems(TestCase):
-    """Tests for API endpoint `db`"""
-    example_items: List[Any]
+        with self.subTest():
+            self.assertFalse(response['success'])
+        with self.subTest():
+            self.assertIn(
+                'can not be greater than 50',
+                response['error']['message'])
 
-    def setUp(self) -> None:
-        self.example_items: List[Any] = [{
-            '_id': uuid4(),
-            'string': 'match me',
-            'integer': 1,
-            'json': {},
-        }, {
-            '_id': uuid4(),
-            'string': 'match me',
-            'integer': 1,
-            'json': {},
-        }, {
-            '_id': uuid4(),
-            'string': "don't match me",
-            'integer': 1,
-            'json': {},
-        }]
-
-        def example_item_query(item: Any) -> psycopg2.sql.Composed:
-            base = sql.SQL(
-                'INSERT INTO example_item (_id, string, integer, json) '
-                'VALUES ({_id}, {string}, {integer}, {json});')
-            query = base.format(
-                _id=sql.Literal(str(item['_id'])),
-                string=sql.Literal(item['string']),
-                integer=sql.Literal(item['integer']),
-                json=sql.Literal(item['json']))
-
-            return query
-
-        # tell psycopg2 to adapt all dictionaries to json instead of
-        # the default hstore
-        register_adapter(dict, Json)
+    def test_response_can_be_paginated_with_limit_and_offset(self) -> None:
+        first_ten = client.call('transaction.get', {'count': 10})['data']
+        next_ten = client.call(
+            'transaction.get', {
+                'count': 10, 'offset': 10})['data']
+        next_ten_ids = [self.trim_msg_id(tran['_id']) for tran in next_ten]
 
-        conn = psycopg2.connect('postgres://test:pass@localhost/dev')
+        print(next_ten_ids)
```

Comment:
Do we want to remove our print lines before PR reviews? If so maybe it needs to be moved over to logging instead.

Reply:
I think this ones fine since it’s in a test and helpful for debugging, but in source code I would definitely say we should remove all print statements or replace them with logging calls.